core/bin/dust_api.rs

```diff
@@ -1277,6 +1206,28 @@ async fn data_sources_documents_update_tags(
     }
 }
 
+fn error_response(
+    status: StatusCode,
+    code: &str,
+    message: &str,
+    error: Option<anyhow::Error>,
+) -> (StatusCode, Json<APIResponse>) {
+    utils::error(&format!("{}: {}\nError: {:?}", code, message, error));
+    (
+        status,
+        Json(APIResponse {
+            error: Some(APIError {
+                code: code.to_string(),
+                message: match error {
+                    Some(err) => format!("{}\nError: {:?}", message, err),
```

Comment:
Let's not have `\n` in our error message: `"{} (error: {:?})"` for now once we have proper logging we might consider once we're happy with it removing it altogether